index.js

```diff
@@ -119,243 +78,174 @@ class CssWriter {
 	}
 }
 
-module.exports = function svelte(options = {}) {
+/** @returns {import('rollup').Plugin} */
+module.exports = function (options = {}) {
+	const extensions = options.extensions || ['.svelte'];
 	const filter = createFilter(options.include, options.exclude);
 
-	const extensions = options.extensions || ['.html', '.svelte'];
-
-	const fixed_options = {};
-
-	Object.keys(options).forEach(key => {
-		// add all options except include, exclude, extensions, and shared
-		if (pluginOptions[key]) return;
-		fixed_options[key] = options[key];
-	});
-
-	if (major_version >= 3) {
-		fixed_options.format = 'esm';
-		fixed_options.sveltePath = options.sveltePath || 'svelte';
-	} else {
-		fixed_options.format = 'es';
-		fixed_options.shared = require.resolve(options.shared || 'svelte/shared.js');
-	}
+	/** @type {import('svelte/types/compiler/interfaces').ModuleFormat} */
+	const format = 'esm', config = { format };
 
-	// handle CSS extraction
-	if ('css' in options) {
-		if (typeof options.css !== 'function' && typeof options.css !== 'boolean') {
-			throw new Error('options.css must be a boolean or a function');
-		}
+	for (let key in options) {
+		// forward `svelte/compiler` options
```

Comment:
I wonder if, given we're releasing a major, we should expect an explicit `compilerOption` property to align with the structure of svelte.config.js?

Reply:
I have no objection to that, but that would be separate issue/PR imo – this PR is breaking for _removing_ existing behavior, not expecting new ones.

Reply:
fair. i'll raise a new issue